sync-gateway-config.json

```diff
@@ -125,10 +125,11 @@ function(doc, oldDoc){
     // Add doc to task-list and moderators channel.
     channel("task-list." + doc.taskList.id);
     channel("moderators");
-  } else if (doc.type == "task-list.user") {
+  } else if (doc.type == "task-list.user" || (oldDoc && oldDoc.type == "task-list.user")) {
```

Comment:
👍  In this case I agree the inner function makes a lot of sense. It's hard to read that one-liner and would be easier to keep it in a single function.

PS: Comparatively, I think the try/catch block (requireUserOrModerator) is still relatively easy to read but if it becomes too much then we'll wrap that one too.